### DISCUSSION TITLE
Secrets Manager

### DISCUSSION THREAD

**Discussion by mortee** [dani-garcia/vaultwarden#3278] [Ideas]
Is there any consideration to look into the [Bitwarden Secrets Manager beta](https://community.bitwarden.com/t/join-the-secrets-manager-beta-waitlist-calling-all-devs-devops/38975), and maybe make VW compatible with it? They are apparently coming out with it real soon now.

**Comment by BlackDex** [comment]
No, this is a Bitwarden licensed feature, not an OpenSource Software/GPLv3 licensed feature and easily integrated

**Reply by paulius-valiunas** [reply]
Isn't Bitwarden itself also Bitwarden licensed? I don't think anyone's asking you to just steal Bitwarden's code, I think the idea is that just how Bitwarden was re-written in Rust to get Vaultwarden, Bitwarden Secrets Manager could be re-written to get a Vaultwarden Secrets Manager. Yes, it's a lot of work, and yes, the answer is probably still the same (no), but I think this deserves a better explanation (i.e. too much effort to implement)

**Reply by BlackDex** [reply]
The frontend which is build-in into the web-vault can't be used. We would have to build that from scratch to and somehow provide the same kind of interface. Which, for one, is not any of the main developers/contributors strong suite of a language (as far as i know)

This project's main focus is on the OpenSource available components of Bitwarden, not the Bitwarden Licensed features.
We have our hands full already on those items.

And the secrets manager is also something handled on the client-side regarding decryption and encryption, access rights, etc..
We would have to build everything including a front-end. That is just too much.

So items which are located here: https://github.com/bitwarden/clients/tree/main/bitwarden_license/bit-web/src/app
Ar not going to be part of Vaultwarden in an easy way. Looking at the whole SSO PR, it also has to jump through some hoops to get it working, with no way to configure it in a nice way per organization, as you would be able to using the licensed part.

If you really want that feature, go to Bitwarden.com and order it there :).

Also deserving a better explanation is a bit of harsh in my opinion.

**Reply by BlackDex** [reply]
Also, no Bitwarden is Licensed the same as Vaultwarden AGPLv3.
Only a few components are under a special Bitwarden License.

**Reply by paulius-valiunas** [reply]
Got it. I did not mean to be harsh, I was just trying to say their license doesn't prevent you from rewriting the backend components for secret management. I don't expect you to do it, but I think what you wrote above is a much more meaningful explanation.

By the way, unless I'm missing something (again), the CLI can be pointed at self-hosted instances so we should be able to use it with Vaultwarden if it supported the secrets API. This means you wouldn't actually have to develop anything client-side.

**Reply by BlackDex** [reply]
Sure, we can build a different Password Manager also haha. That would be a bit stupid i think.
Well, in theory, nothing prevents you from building it your self. You are allowed to use it personally. We are just not allowed to distribute anything of it, if i read the license correctly.
